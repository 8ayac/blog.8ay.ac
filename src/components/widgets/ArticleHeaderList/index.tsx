import { ArticleHeader } from '@/src/components/widgets/ArticleHeader';
import { Article } from '@/src/types';

import styled from '@emotion/styled';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled.div`
    display: grid;
    grid-gap: 5%;
    margin: 0 auto;
  `,

  ArticleHeaderWrapperDiv: styled.div`
    h1 {
      font-size: 2.3rem;
    }
  `,
};

export const ArticleHeaderList: React.FC<{ articles: Article[] }> = ({
  articles,
}) => (
  <>
    <S.ComponentWrapperDiv>
      <S.ArticleHeaderWrapperDiv>
        {articles
          .sort(
            (a, b) =>
              new Date(a.publishedAt).getTime() -
              new Date(b.publishedAt).getTime()
          )
          .reverse()
          .map((article) => (
            <ArticleHeader key={article.attributes.id} article={article} />
          ))}
      </S.ArticleHeaderWrapperDiv>
    </S.ComponentWrapperDiv>
  </>
);
