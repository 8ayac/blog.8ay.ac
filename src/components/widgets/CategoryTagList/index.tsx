import { CategoryTag } from '@/src/components/elements/CategoryTag';
import { FlexContainerDiv } from '@/src/shared/styles/abstractStyledComponents';
import styled from '@emotion/styled';
import { faTags } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';

const S = {
  ComponentWrapperDiv: styled(FlexContainerDiv)`
    flex-flow: row wrap;

    > * {
      margin-right: 0.7rem;
      margin-bottom: 0.7rem;
    }
  `,

  PreIconWrapperDiv: styled(FlexContainerDiv)`
    align-items: center;
    color: ${(props) => props.theme.color.text.primaryLight};
  `,
};

export const CategoryTagList: React.FC<{
  tags: string[];
  withPreIcon?: boolean;
}> = ({ tags, withPreIcon = false }) => (
  <>
    <S.ComponentWrapperDiv>
      {withPreIcon && (
        <S.PreIconWrapperDiv>
          <FontAwesomeIcon icon={faTags} />
        </S.PreIconWrapperDiv>
      )}
      {[...new Set(tags)].sort().map((tag) => (
        <CategoryTag key={tag} name={tag} />
      ))}
    </S.ComponentWrapperDiv>
  </>
);
