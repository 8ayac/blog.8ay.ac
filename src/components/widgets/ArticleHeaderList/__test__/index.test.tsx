import { ArticleHeaderList } from '@/src/components/widgets/ArticleHeaderList';
import { theme } from '@/src/constants/theme';
import { Article } from '@/src/types';
import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';
import React from 'react';

describe('ArticleHeaderList', () => {
  const testArticles = [
    {
      id: 'test-article-01',
      title: 'Test Title 01',
      tags: ['test01'],
      publishedAt: new Date('2021-01-01T00:00:00.000Z'),
      updatedAt: new Date('2021-01-01T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-02',
      title: 'Test Title 02',
      tags: ['test02'],
      publishedAt: new Date('2021-02-02T00:00:00.000Z'),
      updatedAt: new Date('2021-02-02T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-03',
      title: 'Test Title 03',
      tags: ['test03'],
      publishedAt: new Date('2021-03-03T00:00:00.000Z'),
      updatedAt: new Date('2021-03-03T00:00:00.000Z'),
      body: 'bluhbluhbluh',
    },
  ];

  describe('is rendered correctly', () => {
    test('to match the snapshot', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleHeaderList articles={testArticles} />
        </ThemeProvider>
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('to contain the correct ArticleHeader components', () => {
      const wrapper = mount(
        <ThemeProvider theme={theme}>
          <ArticleHeaderList articles={testArticles} />
        </ThemeProvider>
      );

      expect(
        wrapper.find('ArticleHeaderWrapperDiv').find('ArticleHeader')
      ).toHaveLength(3);
      expect(
        (wrapper.find('ArticleHeader').at(0).prop('article') as Article).id
      ).toBe('test-article-03');
      expect(
        (wrapper.find('ArticleHeader').at(1).prop('article') as Article).id
      ).toBe('test-article-02');
      expect(
        (wrapper.find('ArticleHeader').at(2).prop('article') as Article).id
      ).toBe('test-article-01');
    });
  });

  describe('has proper style rules', () => {
    const wrapper = mount(
      <ThemeProvider theme={theme}>
        <ArticleHeaderList articles={testArticles} />
      </ThemeProvider>
    );

    test('in ComponentWrapperDiv', () => {
      expect(wrapper.find('ComponentWrapperDiv')).toHaveStyleRule(
        'margin',
        '0 auto'
      );

      expect(wrapper.find('ComponentWrapperDiv')).toHaveStyleRule(
        'display',
        'grid'
      );

      expect(wrapper.find('ComponentWrapperDiv')).toHaveStyleRule(
        'grid-gap',
        '5%'
      );
    });

    test('in ArticleHeaderWrapperDiv', () => {
      expect(
        wrapper.find('ArticleHeaderWrapperDiv')
      ).toHaveStyleRule('font-size', '2.3rem', { target: 'h1' });
    });
  });
});
