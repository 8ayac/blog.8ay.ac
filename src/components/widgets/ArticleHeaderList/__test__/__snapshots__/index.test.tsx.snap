// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleHeaderList is rendered correctly to match the snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-gap: 5%;
  margin: 0 auto;
}

.emotion-2 h1 {
  font-size: 2.3rem;
}

.emotion-4 {
  margin-bottom: 3rem;
  border-bottom: 0.25rem solid hsla(0, 0%, 75%);
}

.emotion-6 {
  margin-bottom: 0.3rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 1rem;
}

.emotion-8>* {
  margin-right: 1rem;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row wrap;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.emotion-12 {
  margin-right: 0.5rem;
  font-weight: 600;
}

.emotion-14 {
  font-weight: bolder;
  color: hsla(212, 72%, 40%);
}

.emotion-22 {
  margin: 0 0 1rem;
  font-size: 2.16em;
}

.emotion-24 {
  margin: 0;
  color: hsla(0, 0%, 15%);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-26 {
  margin-top: 0.3rem;
}

.emotion-28 {
  margin-top: 2rem;
  margin-bottom: 0.5rem;
}

.emotion-30 {
  margin-right: 1rem;
  margin-left: 0.5rem;
  font-size: 1em;
  color: hsla(0, 0%, 40%);
}

.emotion-32 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0 0.5rem 0.5rem 0;
}

.emotion-34 {
  padding: 3% 1rem 4%;
  margin: 0;
  font-size: 0.8em;
  font-weight: 600;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: hsla(212, 72%, 50%);
  border-radius: 3px;
}

.emotion-34:hover {
  color: white;
  background-color: hsla(212, 72%, 60%);
}

<ArticleHeaderList
  articles={
    Array [
      Object {
        "body": "bluhbluhbluh",
        "id": "test-article-01",
        "publishedAt": "2021-01-01 00:00:00 +0000",
        "tags": Array [
          "test01",
        ],
        "title": "Test Title 01",
        "updatedAt": "2021-01-01 00:00:00 +0000",
      },
      Object {
        "body": "bluhbluhbluh",
        "id": "test-article-02",
        "publishedAt": "2021-02-02 00:00:00 +0000",
        "tags": Array [
          "test02",
        ],
        "title": "Test Title 02",
        "updatedAt": "2021-02-02 00:00:00 +0000",
      },
      Object {
        "body": "bluhbluhbluh",
        "id": "test-article-03",
        "publishedAt": "2021-03-03 00:00:00 +0000",
        "tags": Array [
          "test03",
        ],
        "title": "Test Title 03",
        "updatedAt": "2021-03-03 00:00:00 +0000",
      },
    ]
  }
>
  <ComponentWrapperDiv>
    <div
      className="emotion-0 emotion-1"
    >
      <ArticleHeaderWrapperDiv>
        <div
          className="emotion-2 emotion-3"
        >
          <ArticleHeader
            article={
              Object {
                "body": "bluhbluhbluh",
                "id": "test-article-01",
                "publishedAt": "2021-01-01 00:00:00 +0000",
                "tags": Array [
                  "test01",
                ],
                "title": "Test Title 01",
                "updatedAt": "2021-01-01 00:00:00 +0000",
              }
            }
            key="test-article-01"
          >
            <BodyHeader
              afterTitle={
                <CategoryListWrapperDiv>
                  <FontAwesomeIcon
                    css="unknown styles"
                    icon={
                      Object {
                        "icon": Array [
                          640,
                          512,
                          Array [],
                          "f02c",
                          "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                        ],
                        "iconName": "tags",
                        "prefix": "fas",
                      }
                    }
                  />
                  <CategoryTagList
                    tags={
                      Array [
                        "test01",
                      ]
                    }
                  />
                </CategoryListWrapperDiv>
              }
              beforeTitle={
                <DateTimeWrapperDiv>
                  <DateTime
                    date="2021-01-01 00:00:00 +0000"
                    description="Published At"
                  />
                  <DateTime
                    date="2021-01-01 00:00:00 +0000"
                    description="Updated At"
                  />
                </DateTimeWrapperDiv>
              }
              linkTo="/articles/2021-01-01/test-article-01"
              title="Test Title 01"
            >
              <WrapperHeader>
                <header
                  className="emotion-4 emotion-5"
                >
                  <BeforeTitleDiv>
                    <div
                      className="emotion-6 emotion-7"
                    >
                      <DateTimeWrapperDiv>
                        <div
                          className="emotion-8 emotion-9"
                        >
                          <DateTime
                            date="2021-01-01 00:00:00 +0000"
                            description="Published At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Published At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-01-01T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                          <DateTime
                            date="2021-01-01 00:00:00 +0000"
                            description="Updated At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Updated At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-01-01T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                        </div>
                      </DateTimeWrapperDiv>
                    </div>
                  </BeforeTitleDiv>
                  <TitleH1>
                    <h1
                      className="emotion-22 emotion-23"
                    >
                      <Link
                        href="/articles/2021-01-01/test-article-01"
                        passHref={true}
                      >
                        <TitleAnchor
                          href="/articles/2021-01-01/test-article-01"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          <a
                            className="emotion-24 emotion-25"
                            href="/articles/2021-01-01/test-article-01"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            Test Title 01
                          </a>
                        </TitleAnchor>
                      </Link>
                    </h1>
                  </TitleH1>
                  <AfterTitleDiv>
                    <div
                      className="emotion-26 emotion-27"
                    >
                      <CategoryListWrapperDiv>
                        <div
                          className="emotion-28 emotion-29"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className="emotion-30"
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  640,
                                  512,
                                  Array [],
                                  "f02c",
                                  "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                ],
                                "iconName": "tags",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-tags fa-w-20 emotion-30"
                              data-icon="tags"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 640 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                          <CategoryTagList
                            tags={
                              Array [
                                "test01",
                              ]
                            }
                          >
                            <ComponentWrapperDiv
                              key="test01"
                            >
                              <div
                                className="emotion-32 emotion-33"
                              >
                                <CategoryTag
                                  name="test01"
                                >
                                  <Link
                                    href="/category/test01"
                                    passHref={true}
                                  >
                                    <Anchor
                                      href="/category/test01"
                                      onClick={[Function]}
                                      onMouseEnter={[Function]}
                                    >
                                      <a
                                        className="emotion-34 emotion-35"
                                        href="/category/test01"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                      >
                                        test01
                                      </a>
                                    </Anchor>
                                  </Link>
                                </CategoryTag>
                              </div>
                            </ComponentWrapperDiv>
                          </CategoryTagList>
                        </div>
                      </CategoryListWrapperDiv>
                    </div>
                  </AfterTitleDiv>
                </header>
              </WrapperHeader>
            </BodyHeader>
          </ArticleHeader>
          <ArticleHeader
            article={
              Object {
                "body": "bluhbluhbluh",
                "id": "test-article-02",
                "publishedAt": "2021-02-02 00:00:00 +0000",
                "tags": Array [
                  "test02",
                ],
                "title": "Test Title 02",
                "updatedAt": "2021-02-02 00:00:00 +0000",
              }
            }
            key="test-article-02"
          >
            <BodyHeader
              afterTitle={
                <CategoryListWrapperDiv>
                  <FontAwesomeIcon
                    css="unknown styles"
                    icon={
                      Object {
                        "icon": Array [
                          640,
                          512,
                          Array [],
                          "f02c",
                          "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                        ],
                        "iconName": "tags",
                        "prefix": "fas",
                      }
                    }
                  />
                  <CategoryTagList
                    tags={
                      Array [
                        "test02",
                      ]
                    }
                  />
                </CategoryListWrapperDiv>
              }
              beforeTitle={
                <DateTimeWrapperDiv>
                  <DateTime
                    date="2021-02-02 00:00:00 +0000"
                    description="Published At"
                  />
                  <DateTime
                    date="2021-02-02 00:00:00 +0000"
                    description="Updated At"
                  />
                </DateTimeWrapperDiv>
              }
              linkTo="/articles/2021-02-02/test-article-02"
              title="Test Title 02"
            >
              <WrapperHeader>
                <header
                  className="emotion-4 emotion-5"
                >
                  <BeforeTitleDiv>
                    <div
                      className="emotion-6 emotion-7"
                    >
                      <DateTimeWrapperDiv>
                        <div
                          className="emotion-8 emotion-9"
                        >
                          <DateTime
                            date="2021-02-02 00:00:00 +0000"
                            description="Published At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Published At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-02-02T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                          <DateTime
                            date="2021-02-02 00:00:00 +0000"
                            description="Updated At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Updated At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-02-02T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                        </div>
                      </DateTimeWrapperDiv>
                    </div>
                  </BeforeTitleDiv>
                  <TitleH1>
                    <h1
                      className="emotion-22 emotion-23"
                    >
                      <Link
                        href="/articles/2021-02-02/test-article-02"
                        passHref={true}
                      >
                        <TitleAnchor
                          href="/articles/2021-02-02/test-article-02"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          <a
                            className="emotion-24 emotion-25"
                            href="/articles/2021-02-02/test-article-02"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            Test Title 02
                          </a>
                        </TitleAnchor>
                      </Link>
                    </h1>
                  </TitleH1>
                  <AfterTitleDiv>
                    <div
                      className="emotion-26 emotion-27"
                    >
                      <CategoryListWrapperDiv>
                        <div
                          className="emotion-28 emotion-29"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className="emotion-30"
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  640,
                                  512,
                                  Array [],
                                  "f02c",
                                  "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                ],
                                "iconName": "tags",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-tags fa-w-20 emotion-30"
                              data-icon="tags"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 640 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                          <CategoryTagList
                            tags={
                              Array [
                                "test02",
                              ]
                            }
                          >
                            <ComponentWrapperDiv
                              key="test02"
                            >
                              <div
                                className="emotion-32 emotion-33"
                              >
                                <CategoryTag
                                  name="test02"
                                >
                                  <Link
                                    href="/category/test02"
                                    passHref={true}
                                  >
                                    <Anchor
                                      href="/category/test02"
                                      onClick={[Function]}
                                      onMouseEnter={[Function]}
                                    >
                                      <a
                                        className="emotion-34 emotion-35"
                                        href="/category/test02"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                      >
                                        test02
                                      </a>
                                    </Anchor>
                                  </Link>
                                </CategoryTag>
                              </div>
                            </ComponentWrapperDiv>
                          </CategoryTagList>
                        </div>
                      </CategoryListWrapperDiv>
                    </div>
                  </AfterTitleDiv>
                </header>
              </WrapperHeader>
            </BodyHeader>
          </ArticleHeader>
          <ArticleHeader
            article={
              Object {
                "body": "bluhbluhbluh",
                "id": "test-article-03",
                "publishedAt": "2021-03-03 00:00:00 +0000",
                "tags": Array [
                  "test03",
                ],
                "title": "Test Title 03",
                "updatedAt": "2021-03-03 00:00:00 +0000",
              }
            }
            key="test-article-03"
          >
            <BodyHeader
              afterTitle={
                <CategoryListWrapperDiv>
                  <FontAwesomeIcon
                    css="unknown styles"
                    icon={
                      Object {
                        "icon": Array [
                          640,
                          512,
                          Array [],
                          "f02c",
                          "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                        ],
                        "iconName": "tags",
                        "prefix": "fas",
                      }
                    }
                  />
                  <CategoryTagList
                    tags={
                      Array [
                        "test03",
                      ]
                    }
                  />
                </CategoryListWrapperDiv>
              }
              beforeTitle={
                <DateTimeWrapperDiv>
                  <DateTime
                    date="2021-03-03 00:00:00 +0000"
                    description="Published At"
                  />
                  <DateTime
                    date="2021-03-03 00:00:00 +0000"
                    description="Updated At"
                  />
                </DateTimeWrapperDiv>
              }
              linkTo="/articles/2021-03-03/test-article-03"
              title="Test Title 03"
            >
              <WrapperHeader>
                <header
                  className="emotion-4 emotion-5"
                >
                  <BeforeTitleDiv>
                    <div
                      className="emotion-6 emotion-7"
                    >
                      <DateTimeWrapperDiv>
                        <div
                          className="emotion-8 emotion-9"
                        >
                          <DateTime
                            date="2021-03-03 00:00:00 +0000"
                            description="Published At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Published At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-03-03T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                          <DateTime
                            date="2021-03-03 00:00:00 +0000"
                            description="Updated At"
                          >
                            <ComponentWrapperDiv>
                              <div
                                className="emotion-10 emotion-11"
                              >
                                <DescriptionSpan>
                                  <span
                                    className="emotion-12 emotion-13"
                                  >
                                    Updated At:
                                  </span>
                                </DescriptionSpan>
                                <DateTime>
                                  <time
                                    className="emotion-14 emotion-15"
                                  >
                                    2021-03-03T00:00:00.000Z
                                  </time>
                                </DateTime>
                              </div>
                            </ComponentWrapperDiv>
                          </DateTime>
                        </div>
                      </DateTimeWrapperDiv>
                    </div>
                  </BeforeTitleDiv>
                  <TitleH1>
                    <h1
                      className="emotion-22 emotion-23"
                    >
                      <Link
                        href="/articles/2021-03-03/test-article-03"
                        passHref={true}
                      >
                        <TitleAnchor
                          href="/articles/2021-03-03/test-article-03"
                          onClick={[Function]}
                          onMouseEnter={[Function]}
                        >
                          <a
                            className="emotion-24 emotion-25"
                            href="/articles/2021-03-03/test-article-03"
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            Test Title 03
                          </a>
                        </TitleAnchor>
                      </Link>
                    </h1>
                  </TitleH1>
                  <AfterTitleDiv>
                    <div
                      className="emotion-26 emotion-27"
                    >
                      <CategoryListWrapperDiv>
                        <div
                          className="emotion-28 emotion-29"
                        >
                          <FontAwesomeIcon
                            border={false}
                            className="emotion-30"
                            fixedWidth={false}
                            flip={null}
                            icon={
                              Object {
                                "icon": Array [
                                  640,
                                  512,
                                  Array [],
                                  "f02c",
                                  "M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z",
                                ],
                                "iconName": "tags",
                                "prefix": "fas",
                              }
                            }
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            swapOpacity={false}
                            symbol={false}
                            title=""
                            transform={null}
                          >
                            <svg
                              aria-hidden="true"
                              className="svg-inline--fa fa-tags fa-w-20 emotion-30"
                              data-icon="tags"
                              data-prefix="fas"
                              focusable="false"
                              role="img"
                              style={Object {}}
                              viewBox="0 0 640 512"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M497.941 225.941L286.059 14.059A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v204.118a48 48 0 0 0 14.059 33.941l211.882 211.882c18.744 18.745 49.136 18.746 67.882 0l204.118-204.118c18.745-18.745 18.745-49.137 0-67.882zM112 160c-26.51 0-48-21.49-48-48s21.49-48 48-48 48 21.49 48 48-21.49 48-48 48zm513.941 133.823L421.823 497.941c-18.745 18.745-49.137 18.745-67.882 0l-.36-.36L527.64 323.522c16.999-16.999 26.36-39.6 26.36-63.64s-9.362-46.641-26.36-63.64L331.397 0h48.721a48 48 0 0 1 33.941 14.059l211.882 211.882c18.745 18.745 18.745 49.137 0 67.882z"
                                fill="currentColor"
                                style={Object {}}
                              />
                            </svg>
                          </FontAwesomeIcon>
                          <CategoryTagList
                            tags={
                              Array [
                                "test03",
                              ]
                            }
                          >
                            <ComponentWrapperDiv
                              key="test03"
                            >
                              <div
                                className="emotion-32 emotion-33"
                              >
                                <CategoryTag
                                  name="test03"
                                >
                                  <Link
                                    href="/category/test03"
                                    passHref={true}
                                  >
                                    <Anchor
                                      href="/category/test03"
                                      onClick={[Function]}
                                      onMouseEnter={[Function]}
                                    >
                                      <a
                                        className="emotion-34 emotion-35"
                                        href="/category/test03"
                                        onClick={[Function]}
                                        onMouseEnter={[Function]}
                                      >
                                        test03
                                      </a>
                                    </Anchor>
                                  </Link>
                                </CategoryTag>
                              </div>
                            </ComponentWrapperDiv>
                          </CategoryTagList>
                        </div>
                      </CategoryListWrapperDiv>
                    </div>
                  </AfterTitleDiv>
                </header>
              </WrapperHeader>
            </BodyHeader>
          </ArticleHeader>
        </div>
      </ArticleHeaderWrapperDiv>
    </div>
  </ComponentWrapperDiv>
</ArticleHeaderList>
`;
